instrumentation/dalli/test/opentelemetry/instrumentation/dalli/instrumentation_test.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/instrumentation/dalli'
+require_relative '../../../../lib/opentelemetry/instrumentation/dalli/patches/server'
+
+describe OpenTelemetry::Instrumentation::Dalli::Instrumentation do
+  let(:instrumentation) { OpenTelemetry::Instrumentation::Dalli::Instrumentation.instance }
+  let(:exporter) { EXPORTER }
+  let(:span) { exporter.finished_spans.first }
+  let(:dalli) { ::Dalli::Client.new(nil, {}) }
+
+  before do
+    exporter.reset
+  end
+
+  after do
+    # Force re-install of instrumentation
+    instrumentation.instance_variable_set(:@installed, false)
+  end
+
+  describe 'tracing' do
+    before do
+      instrumentation.install
+    end
+
+    it 'before request' do
+      _(exporter.finished_spans.size).must_equal 0
+    end
+
+    it 'after dalli#set' do
+      dalli.set('foo', 'bar')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'SET'
+      _(span.attributes['db.system']).must_equal 'memcached'
+      _(span.attributes['db.statement']).must_equal 'set foo bar 0 0'
+      _(span.attributes['net.peer.name']).must_equal '127.0.0.1'
+      _(span.attributes['net.peer.port']).must_equal 11211
+    end
+
+    it 'after dalli#get_multi' do
+      dalli.get_multi('foo', 'bar')
+
+      _(exporter.finished_spans.size).must_equal 1
+      _(span.name).must_equal 'SEND_MULTIGET'
```

Comment:
🤔 I'd prefer to see actual `memcache` commands here. IIRC, this should be a `GET`. For cases where Dalli maps the command names to something else, it'd be good to map them back to the real command names.